src/constraint/encoding.ts

```diff
@@ -221,8 +222,10 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
     strict: true,
     satisfy: (encQ: EncodingQuery, schema: Schema, opt: QueryConfig) => {
       if (encQ.scale) {
-        const scaleType = (encQ.scale as ScaleQuery).type;
-        const type = encQ.type;
+        const sType = (encQ.scale as ScaleQuery).type as ScaleType;
+        const timeUnit = encQ.timeUnit as TimeUnit;
+        const type = encQ.type as Type;
+        const scaleType = _scaleType(sType, timeUnit, type);
```

Comment:
1) Unnecessary declarations, why don't you pass these parameters directly to the methods?
2) I commented that use _scaleType as local variable (In hindsight, `sType` might be better) and keep the method called `scaleType`.